src/proxyvisor.coffee

```diff
@@ -0,0 +1,243 @@
+Promise = require 'bluebird'
+dockerUtils = require './docker-utils'
+{ docker } = dockerUtils
+express = require 'express'
+fs = Promise.promisifyAll require 'fs'
+{ resinApi } = require './request'
+knex = require './db'
+_ = require 'lodash'
+deviceRegister = require 'resin-register-device'
+randomHexString = require './lib/random-hex-string'
+utils = require './utils'
+device = require './device'
+bodyParser = require 'body-parser'
+request = Promise.promisifyAll require 'request'
+config = require './config'
+PUBNUB = require 'pubnub'
+execAsync = Promise.promisify(require('child_process').exec)
+
+pubnub = PUBNUB.init(config.pubnub)
+
+getAssetsPath = (image) ->
+	docker.imageRootDir(image)
+	.then (rootDir) ->
+		return rootDir + '/assets'
+
+exports.router = router = express.Router()
+router.use(bodyParser())
+
+router.get '/v1/devices', (req, res) ->
+	knex('dependentDevice').select()
+	.then (devices) ->
+		res.json(devices)
+	.catch (err) ->
+		res.status(503).send(err?.message or err or 'Unknown error')
+
+router.post '/v1/devices', (req, res) ->
+	Promise.join(
+		utils.getConfig('apiKey')
+		utils.getConfig('userId')
+		device.getID()
+		deviceRegister.generateUUID()
+		randomHexString.generate()
+		(apiKey, userId, deviceId, uuid, logsChannel) ->
+			d =
+				user: userId
+				application: req.body.applicationId
+				uuid: uuid
+				device_type: 'edge'
+				device: deviceId
+				registered_at: Math.floor(Date.now() / 1000)
+				logs_channel: logsChannel
+				status: 'Provisioned'
+			resinApi.post
```

Comment:
@Page- any ideas on how to reconcile this with #224 ?
Maybe add a dependent device register in resin-io-modules/resin-register-device?